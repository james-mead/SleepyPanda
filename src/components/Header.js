import React, { Component } from 'react'
import {
  Text,
  View,
  Image,
  StyleSheet } from 'react-native'

export default class Header extends Component {
  render () {
    return (
      <View style={style.header}>
        <Image source={require('../images/Logo.png')} style={style.appLogo} />
        <Text style={style.appTitle}>
          Sleepy Panda
        </Text>
      </View>
    )
  }
}

const style = StyleSheet.create({
  header: {
    backgroundColor: '#3D5667',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    paddingTop: 20,
    padding: 10
  },
  appTitle: {
    color: '#f0f8ff',
    fontFamily: 'AmericanTypewriter-Bold',
    fontSize: 10,
    letterSpacing: 1.5,
    textAlign: 'left'
  },
  appLogo: {
    height: 40,
    width: 60,
    backgroundColor: 'transparent'
  }
})
