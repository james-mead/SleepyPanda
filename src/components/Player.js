import React, { Component } from 'react'
import {
  View,
  Image,
  StyleSheet,
  TouchableHighlight } from 'react-native'
import Icon from 'react-native-vector-icons/FontAwesome'

export default class Player extends Component {
  render () {
    return (
      <View style={style.container}>
        <View style={style.imageContainer}>
          <Image source={this.props.image} style={style.image} />
        </View>
        {this.props.status
        ? <TouchableHighlight underlayColor="white" onPress={() => { pauseSound() }}>
            <Icon className="icon" name="pause" size={30} color="#FFF" />
          </TouchableHighlight>
        : <TouchableHighlight underlayColor="white" onPress={() => { playSound() }}>
            <Icon className="icon" name="play" size={30} color="#FFF" />
          </TouchableHighlight>
        }
      </View>
    )
  }
}

const style = StyleSheet.create({
  container: {
    flex: 1,
    // flexWrap: 'wrap',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#7A8B98',
    borderTopWidth: StyleSheet.hairlineWidth,
    borderTopColor: '#FFFFFF',
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    padding: 10,
    height: 100
  },
  imageContainer: {
    backgroundColor: 'gray',
    borderRadius: 10,
    position: 'absolute',
    left: 10,
    alignItems: 'center',
    height: 70,
    width: 70,
    shadowColor: '#000000',
    shadowOpacity: 0.8,
    shadowRadius: 2,
    shadowOffset: {
      height: 2,
      width: 2
      }
    },
    image: {
      height: 70,
      width: 70,
      borderRadius: 10
    },
    playButton: {
      alignItems: 'center',
      color: '#f0f8ff'
    }
});
