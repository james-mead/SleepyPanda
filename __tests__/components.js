import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
// import { shallow } from 'enzyme'

import Header from '../src/components/Header.js'
import Player from '../src/components/Player.js'
// import SoundList from '../src/components/SoundList.js'

// function MyComponent() {
//   return <div />
// }
//
// const renderer = createRenderer()
//
// const tree = renderer.render(<MyComponent />)
// expect(isDOMComponent(tree)).toBe(true)

// describe('Foo component', function() {
//   it('renders 3 Image components with the right props', function() {
//     const renderer = TestUtils.createRenderer()
//     renderer.render(<Header />)
//     const result = renderer.getRenderOutput()
//     const sources = map(findAllWithType(result, Image),
//                         'props.source')
//     expect(sources).toDeepEqual([
//       '../images/Logo.png'
//     ])
//   })
// })

describe('Header (Snapshot)', () => {
  it('Renders the Header Component', () => {
    const component = renderer.create(<Header />)
    const json = component.toJSON()
    expect(json).toMatchSnapshot()
  })
})

describe('Player (Snapshot)', () => {
  it('Renders the Player Component', () => {
    const component = renderer.create(<Player />)
    const json = component.toJSON()
    expect(json).toMatchSnapshot()
  })
})

describe('Confirms Sound Data is loaded', () => {
  it('renders the name of each sound file', () => {
  const soundData = require('../src/data/sounds')
  const ListView = require('ListView')
  const Text = require('Text')
  const dataSource = new ListView.DataSource({
    rowHasChanged: (r1, r2) => r1 !== r2,
  }).cloneWithRows(soundData)
  const component = renderer
    .create(
      <ListView
        dataSource={dataSource}
        renderRow={rowData => <Text>{rowData.name}</Text>}
      />
    )
    const json = component.toJSON()
  expect(component).toMatchSnapshot()
})
})
