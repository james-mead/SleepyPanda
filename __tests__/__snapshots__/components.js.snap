// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirms Sound Data is loaded renders the name of each sound file 1`] = `
<RCTScrollView
  dataSource={
    ListViewDataSource {
      "items": 0,
    }
  }
  renderRow={[Function]}
  renderScrollComponent={[Function]}
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
</RCTScrollView>
`;

exports[`Header (Snapshot) Renders the Header Component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#3D5667",
      "flexDirection": "column",
      "justifyContent": "center",
      "padding": 10,
      "paddingTop": 20,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "backgroundColor": "transparent",
        "height": 70,
        "width": 100,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#f0f8ff",
        "fontFamily": "AmericanTypewriter-Bold",
        "fontSize": 10,
        "letterSpacing": 1.5,
        "textAlign": "left",
      }
    }
  >
    Sleepy Panda
  </Text>
</View>
`;

exports[`Player (Snapshot) Renders the Player Component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#7A8B98",
      "borderTopColor": "#FFFFFF",
      "borderTopWidth": 0.5,
      "bottom": 0,
      "flex": 1,
      "flexDirection": "row",
      "height": 100,
      "justifyContent": "center",
      "left": 0,
      "padding": 10,
      "position": "absolute",
      "right": 0,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "gray",
        "borderRadius": 10,
        "height": 70,
        "left": 10,
        "position": "absolute",
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 0.8,
        "shadowRadius": 2,
        "width": 70,
      }
    }
  >
    <Image
      source={undefined}
      style={
        Object {
          "borderRadius": 10,
          "height": 70,
          "width": 70,
        }
      }
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      className="icon"
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 30,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      ÔÅã
    </Text>
  </View>
</View>
`;
